src/pages/programDisaggregations/list/ProgramsList.tsx

```diff
@@ -0,0 +1,218 @@
+import { useAlert, useDataEngine, useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    ButtonStrip,
+    Modal,
+    ModalActions,
+    ModalContent,
+    ModalTitle,
+    NoticeBox,
+} from '@dhis2/ui'
+import React, { useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { LinkButton } from '../../../components/LinkButton'
+import { LoadingSpinner } from '../../../components/loading/LoadingSpinner'
+import { ModelSingleSelect } from '../../../components/metadataFormControls/ModelSingleSelect'
+import { UpdateMutation, useLocationSearchState } from '../../../lib'
+import { Program } from '../../../types/generated'
+import classes from './ProgramsList.module.css'
+
+export interface ProgramsData {
+    results: {
+        programs: Program[]
+    }
+}
+
+const programsQuery = {
+    resource: 'programs',
+    params: {
+        fields: ['displayName', 'id', 'categoryMappings[id]'],
+    },
+}
+
+const programsQueryResults = {
+    results: {
+        resource: 'programs/gist',
+        params: {
+            fields: ['name', 'displayName', 'id', 'categoryMappings'],
+            order: 'name:asc',
+            pageSize: 200,
+            // filter: 'categoryMappings:!empty',
+        },
+    },
+}
+
+export const ProgramsList = () => {
+    const preservedSearchState = useLocationSearchState()
+    const { loading, error, data, refetch } =
+        useDataQuery<ProgramsData>(programsQueryResults)
+    const navigate = useNavigate()
+    const dataEngine = useDataEngine()
+    const saveAlert = useAlert(
+        ({ message }) => message,
+        (options) => options
+    )
+    const [deleteModalOpen, setDeleteModalOpen] = useState(false)
+    const [programToDelete, setProgramToDelete] = useState<Program | null>(null)
```

Comment:
I would've preferred to extract this logic and related halders below to a hook, but not super important.

Reply:
I would also argue only one of these should be needed? Eg. `programToDelete` would control the open-state of the dialog?